### PR TITLE
Connection: add 'heartbeat/data' REST endpoint

### PR DESCRIPTION
**Pull request by sergeymitr**
## Proposed changes:
* Add the REST endpoint to fetch heartbeat data via REST.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pf5801-E9-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
See https://github.com/Automattic/vulcan/issues/204#issuecomment-2016639892
This PR may be tested and merged separately from the diff.

### CODE REVIEW

File: projects/packages/connection/src/class-heartbeat.php [hunk 1]
```diff
@@ -249,4 +256,55 @@ public function cli_callback( $args ) {
...
+
+	/**
+	 * Initialize the heartbeat REST API.
+	 *
+	 * @return void
+	 */
+	public function initialize_rest_api() {
+		register_rest_route(
+			'jetpack/v4',
+			'/get_heartbeat_data',
```
Comment (zinigor) [inline projects/packages/connection/src/class-heartbeat.php]: I haven't reviewed the entire thing yet, but do you think that `get_heartbeat_data` is a redundant name? I don't think snake case is ever used in REST API, and even if it was, we are already specifying that it is a `GET` with an HTTP verb. And `data` is like - what else can you get from an API endpoint? :)

Comment (zinigor) [reply projects/packages/connection/src/class-heartbeat.php]: `/jetpack/v4/heartbeat` looks a lot better to me

Comment (sergeymitr) [reply projects/packages/connection/src/class-heartbeat.php]: Good idea.
I named it `/jetpack/v4/heartbeat/data` to emphasize there's data being retrieved 🙂

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Pull request merged**